doc/man3/OSSL_PROVIDER.pod

```diff
@@ -0,0 +1,110 @@
+=pod
+
+=head1 NAME
+
+OSSL_PROVIDER, OSSL_load_provider, OSSL_unload_provider,
+OSSL_get_provider_param_types, OSSL_get_provider_params,
+OSSL_add_provider - provider routines
+
+=head1 SYNOPSIS
+
+ #include <openssl/provider.h>
+
+ typedef struct ossl_provider_st OSSL_PROVIDER;
+
+ OSSL_PROVIDER *OSSL_load_provider(OPENSSL_CTX *, const char *name);
+ int OSSL_unload_provider(OSSL_PROVIDER *prov);
+
+ const OSSL_ITEM *OSSL_get_provider_param_types(OSSL_PROVIDER *prov);
+ int OSSL_get_provider_params(OSSL_PROVIDER *prov, const OSSL_PARAM params[]);
+
+ int OSSL_add_provider(OPENSSL_CTX *,
+                       const char *name, ossl_provider_init_fn *init_fn);
+
+=head1 DESCRIPTION
+
+C<OSSL_PROVIDER> is a type that holds internal information about
+implementation providers (see L<provider(7)> for information on what a
+provider is).
+A provider can be built in to the application or the OpenSSL
+libraries, or can be a loadable module.
+The functions described here handle both forms.
+
+=head2 Functions
+
+OSSL_add_provider() is used to add a built in provider in the given
+library context, by associating a provider name with a provider
+initialization function.
+This name can then be used with OSSL_load_provider().
+
+OSSL_load_provider() loads and initializes a provider.
+This may simply initialize a provider that was previously added with
+OSSL_add_provider() and simply run its given initialization function,
+or load a provider module with the given name and run its provider
+entry point, C<OSSL_provider_init>.
+
+OSSL_unload_provider() unloads the given provider.
+
+OSSL_get_provider_param_types() is used to get a provider parameter
+descriptor set as an C<OSSL_ITEM> array.
+Each element is a tuple of an C<OSSL_PARAM> parameter type and a name
+in form of a C string.
+See L<openssl-core.h(7)> for more information on C<OSSL_ITEM> and
+parameter types.
+
+OSSL_get_provider_params() is used to get provider parameter values.
+The caller must prepare the C<OSSL_PARAM> array before calling this
+function, and the variables acting as buffers for this parameter array
```

Comment:
Could lose the comma here.

Reply:
I'm unsure.  If you don't have strong feelings about it, I prefer to keep it.

Reply:
Sure, both work.